ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -157,7 +157,18 @@ peerStates Peer {name, value = schedulePoints} =
 --
 -- The resulting schedule contains all the peers. Items are sorted by time.
 peersStates :: PointSchedule blk -> [(Time, Peer (NodeState blk))]
-peersStates peers = foldr (mergeOn fst) [] (peerStates <$> toList (peersList $ unPointSchedule peers))
+peersStates PointSchedule{psSchedule, psMinDuration} =
+  case foldr (mergeOn fst) [] (peerStates <$> peersList psSchedule) of
+    [] -> if psMinDuration > Time 0
+        then case peersList psSchedule of
+            []  -> []
+            p:_ -> [(psMinDuration, (const genesisNodeState) <$> p)]
+        else []
+    peers ->
+      let (t, p) = last peers
+      in if t < psMinDuration
+          then peers ++ [(psMinDuration, p)]
+          else peers
```

Comment:
I thought the whole point was to get rid of these weird additional ticks, so I would have expected to get rid of this function, or at least to not have these lines 168--171. Is this really unavoidable?

Reply:
I now see your comment:

> Dear reviewers, what's your opinion on that? I think the previous option was a better compromise: duplicate the last point "under the hood" so that the experiment lasts for the expected duration.

hinting at this. I do think this is a better compromise than the other one; I just wish we could do a bit better!